src/web/modules/transactions/discrepancies/validateLedger.http.ts

```diff
@@ -13,39 +15,79 @@ export async function validateLedgerTransaction(
 ): Promise<void> {
   try {
     const ledgerEntry = await Ledger.transactions.retrieve(req.params.id)
-    const connectorEntry = await Connector.charges.retrieveAPI(req.params.id, ledgerEntry.gateway_account_id)
+    const connectorEntry = await Connector.charges.parityCheck(req.params.id, ledgerEntry.gateway_account_id)
+    let warningMessage
+    if (typeof connectorEntry === 'string') {
+        warningMessage = connectorEntry
+    }
 
     const ledgerResponseWithoutLedgerSpecificFields = _.omit(ledgerEntry, [
-      'gateway_account_id',
-      'transaction_id',
-      'disputed',
-      'source',
-      'live',
-      'transaction_type',
-      'credential_external_id',
-      'service_id',
-      'refund_summary.amount_refunded'
+    'gateway_account_id',
+    'transaction_id',
+    'disputed',
+    'source',
+    'live',
+    'transaction_type',
+    'credential_external_id',
+    'service_id',
+    'refund_summary.amount_refunded',
+    'evidence_due_date',
+    'gateway_payout_id',
+    'parent_transaction_id',
+    'payment_details',
+    'reason'
     ])
     const connectorResponseWithoutConnectorSpecificFields = _.omit(connectorEntry, [
-        'links',
-        'charge_id'
+    'links',
+    'charge_id',
+    'auth_code',
+    'authorised_date',
+    'payment_outcome',
+    'processor_id',
+    'provider_id',
+    'telephone_number'
     ])
 
     const parity = diff(connectorResponseWithoutConnectorSpecificFields, ledgerResponseWithoutLedgerSpecificFields).filter(obj => {
-        if (obj.path[0] !== 'created_date' && obj.kind === EDIT) {
-            return true
+        switch (obj.kind) {
+            case EDIT: {
+                if (obj.path[0] !== 'created_date') {
+                    return true
+                } else {
+                    const castedDiffObj = obj as DiffEdit<string>
+                    const lhsDate = moment(castedDiffObj.lhs)
+                    const rhsDate = moment(castedDiffObj.rhs)
+                    return Math.abs(lhsDate.diff(rhsDate)) > 10;
```

Comment:
is this diff measuring `ms`? We might want to consider a longer time frame

Reply:
yes, we weren't sure what to go for here. @kbottla - any thoughts on what time difference in creationdate sholud be regarded as a meaningful difference please?

Reply:
This might be in seconds.

We allow a difference of 5 seconds in connector https://github.com/alphagov/pay-connector/blob/a5c48d33fa7f2e54ec6bf49d9dc7c9e56fae4b7c/src/main/java/uk/gov/pay/connector/tasks/service/ChargeParityChecker.java#L134. Although it is for historic charges, we could keep the same.

Reply:
<img width="538" alt="Screenshot 2024-09-19 at 13 04 01" src="https://github.com/user-attachments/assets/8d05b194-4959-4e54-a221-830e6eb348d4">